app/vmalert/datasource/vm_test.go

```diff
@@ -150,6 +150,139 @@ func TestVMInstantQuery(t *testing.T) {
 	}
 }
 
+func TestVMInstantQueryWithAuthorization(t *testing.T) {
```

Comment:
the test looks too excessive and contains a lot of copy-pasted code.
Testing the auth headers can be done in a more straightforward way of just checking the headers. Please, consider https://github.com/VictoriaMetrics/VictoriaMetrics/blob/ef1afeed6c2bd427aee92d15e78c1330926b5af2/app/vmalert/datasource/vm_test.go#L374-L382 as an example

Reply:
removed, implemented extended test